src/app/common/stores/applications/SsvWeb/Operator.store.ts

```diff
@@ -1019,27 +724,7 @@ class OperatorStore extends BaseStore {
           const event: boolean = receipt.hasOwnProperty('events');
           if (event) {
             ApiParams.initStorage(true);
-            console.debug('Contract Receipt', receipt);
-            let iterations = 0;
-            while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
-              // Reached maximum iterations
-              if (iterations >= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
-                // eslint-disable-next-line no-await-in-loop
-                await this.refreshOperatorsAndClusters(resolve, true);
-                break;
-              }
-              iterations += 1;
-              // eslint-disable-next-line no-await-in-loop
-              if (!(await myAccountStore.checkEntityInAccount('operator', 'id', parseInt(String(operatorId), 10)))) {
-                // eslint-disable-next-line no-await-in-loop
-                await this.refreshOperatorsAndClusters(resolve, true);
-                break;
-              } else {
-                console.log('Operator is still in API..');
-              }
-              // eslint-disable-next-line no-await-in-loop
-              await myAccountStore.delay();
-            }
+            await ifEventCaught(receipt.transactionHash, async () => this.refreshOperatorsAndClusters(resolve, true), myAccountStore.delay);
```

Comment:
@IlyaVi @axelrod-blox this is a good example to what I wrote above:

in the entire project, search for the line:
`await ifEventCaught(receipt.transactionHash, async () =>`

It appears 6 times in Validator and operator store, slightly different every time but very simialr. I think this could be generalized with oo design